File: Smart Sweepers/src/utils/Random.java
package utils;

public class Random {

	java.util.Random random;

	public Random() {
		random = new java.util.Random();
	}

	public int randomInt(int x, int y) {
		return random.nextInt(Integer.MAX_VALUE) % ((y - x + 1) + x);
	}

	public double randomFloat() {
		return random.nextDouble();
	}
	
	public double randomFloat(double lower, double upper) {
		double range = upper - lower;
		return lower + random.nextDouble() * range;
	}

	public boolean randomBoolean() {
		return randomInt(0, 1) > 0.5;
	}

	public double randomClamped() {
		return randomFloat() - randomFloat();
	}
}

File: Smart Sweepers/src/utils/Random.kt
package utils

import kotlin.random.*

@Suppress("ClassName")
object rand {
    @JvmStatic
    fun randomFloat() = Random.Default.nextDouble()

    @JvmStatic
    fun randomRadian() = randomFloat() * Math.PI * 2

    @JvmStatic
    fun randomFloat(lower: Double, upper: Double) = Random.Default.nextDouble(lower, upper)

    /** *triangular* (__not__ linear) random number between -1.0 and 1.0, peaking at 0.0 */
    @JvmStatic
    fun randomClamped() = randomFloat() - randomFloat()

    @JvmStatic
    fun randomInt(inclusiveMin: Int, inclusiveMax: Int) = Random.Default.nextInt(inclusiveMin, inclusiveMax + 1)

    @JvmStatic
    fun randomBoolean() = Random.Default.nextBoolean()
}
